lib/portage/sync/syncbase.py

```diff
@@ -303,6 +294,21 @@ def noisy_refresh_keys():
 				loop.run_until_complete(decorated_func())
 		out.eend(0)
 
+	def _set_proxy(self, openpgp_env):
```

Comment:
@mgorny what about doing this way?

Reply:
I've just told you that you're not supposed to touch the attribute on `OpenPGPEnvironment`, so rather some method that creates `OpenPGPEnvironment`. They may be some more stuff worth putting into the common method.

Reply:
> I've just told you that you're not supposed to touch the attribute on `OpenPGPEnvironment`, 

Sorry but I've probably missed that part in the understanding of 'your sentence of helper function in base class'. Also, as `proxy` is not a private attribute that is not clear. I'll redo something in this sense then.